plugin/src/main/java/org/opensearch/ml/searchext/MLInferenceRequestParametersExtBuilder.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.ml.searchext;
+
+import static org.opensearch.ml.searchext.MLInferenceRequestParameters.ML_INFERENCE_FIELD;
+
+import java.io.IOException;
+import java.util.Objects;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.core.xcontent.XContentParser;
+import org.opensearch.search.SearchExtBuilder;
+
+public class MLInferenceRequestParametersExtBuilder extends SearchExtBuilder {
```

Comment:
Please add class level java doc that describes class responsibilities

Reply:
added